### PR TITLE
No download dir for editable

### PR DESCRIPTION
**Pull request by dfee**
A rebase of https://github.com/jazzband/pip-tools/pull/385.

### CODE REVIEW

**Comment by dfee** [comment]
@vphilippon I'd like to get this in the next release as it replaces #385. Can we get this done?

File: tests/test_cli.py [hunk 1]
```diff
@@ -222,6 +222,19 @@ def test_editable_package(tmpdir):
         assert 'six==1.10.0' in out.output
 
 
+def test_editable_package_vcs(tmpdir):
+    vcs_package = 'git+git://github.com/pytest-dev/pytest-django#egg=pytest-django'
```
Comment (vphilippon) [inline tests/test_cli.py]: I would pin this to the current latest commit of `pytest-django` master branch.
The `pytest-django` package could stop using `setuptools_scm`.
Also, if they break the master branch for a while, I'd rather not have our tests fail on that.

File: tests/test_cli.py [hunk 2]
```diff
@@ -222,6 +222,19 @@ def test_editable_package(tmpdir):
...
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        with open('requirements.in', 'w') as req_in:
+            req_in.write('-e ' + vcs_package)
+        out = runner.invoke(cli, ['-n',
+                                  '--rebuild'])
+        print(out.output)
+        assert out.exit_code == 0
+        assert vcs_package in out.output
```
Comment (vphilippon) [inline tests/test_cli.py]: Add a check that `pytest` is also in the output, as it's one of `pytest-django`'s `install_requires`.
We want to make sure that we still collect the subdependencies.

File: tests/test_cli.py (hunk 1 above)
Comment (dfee) [inline tests/test_cli.py]: Sounds good.

File: tests/test_cli.py (hunk 2 above)
Comment (dfee) [inline tests/test_cli.py]: Sounds good.

**Comment by dfee** [comment]
@vphilippon updated.

**Comment by vphilippon** [comment]
Thanks for the contribution!

**Pull request merged**